script/bump-version.js

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env node
+
+const { GitProcess } = require('dugite')
+const utils = require('./lib/version-utils')
+const plist = require('plist')
+const fs = require('fs')
+const semver = require('semver')
+const path = require('path')
+const { promisify } = require('util')
+const minimist = require('minimist')
+
+const writeFile = promisify(fs.writeFile)
+const readFile = promisify(fs.readFile)
+
+function parseCommandLine () {
+  let help
+  const opts = minimist(process.argv.slice(2), {
+    string: [ 'bump', 'version' ],
+    boolean: [ 'dryRun', 'help' ],
+    alias: { 'version': ['v'] },
+    unknown: arg => { help = true }
+  })
+  if (help || opts.help || !opts.bump) {
+    console.log(`
+      Bump release version number. Possible arguments:\n
+        --bump=patch to increment patch version\n
+        --version={version} to set version number directly\n
+        --dryRun to print the next version without updating files
+      Note that you can use both --bump and --stable  simultaneously. 
+    `)
+    process.exit(0)
+  }
+  return opts
+}
+
+// run the script
+async function main () {
+  const opts = parseCommandLine()
+  const currentVersion = await utils.getElectronVersion()
+  const version = await nextVersion(opts.bump, currentVersion)
+  const components = semver.valid(semver.coerce(version)).split('.')
+
+  let pre
+  if (version.includes('-')) {
+    pre = `-${version.split('-')[1]}`
+    components[3] = version.split('-')[1].split('.')[1]
+  }
```

Comment:
I had to read this a few times to understand what it was doing. I have a couple of thoughts.

First, we're passing this `components` array around, but it's unclear what each of the components actually is. We index the array by magic constants everywhere and it's not clear why. I think making this an object with `{ major, minor, patch, prereleaseNum }` or similar would help a lot with readability. This would make it harder to just `components.join(".")` but a helper function could handle that and I think the increase in overall readability would be worth it.

Second, I `semver` exposes a `parse` method that gives you this information without the relatively cryptic string manipulation:

```javascript
> semver.parse('2.0.0-beta.3')
SemVer {
  loose: undefined,
  raw: '2.0.0-beta.3',
  major: 2,
  minor: 0,
  patch: 0,
  prerelease: [ 'beta', 3 ],
  build: [],
  version: '2.0.0-beta.3' }
```